woocommerce/action-scheduler-admin
# Add search feature

This PR adds a simple search control and search functionality comparable to the core admin screen.

### Testing instructions

- Use this snippet to create test pending actions
```
add_action( 'init', function() {
	$prefix = [ 'one', 'two', 'three', 'four', 'five' ];
    for ( $i = 1; $i <= 20; $i++ ) {
		$hook = $prefix[ $i % 5 ] . '-single-test-' . $i;
        as_schedule_single_action( time() + DAY_IN_SECONDS, $hook );
	}
} );
```
- Run `npm run install`
- Run `npm run build`
- Go to WooCommerce -> Scheduled Actions
- Type 'one', 'two', 'three', 'four', or 'five' in the search box
- Click the search icon
- Verify all rows in the table match the search string
- Verify report paging, filtering, and ordering functions work correctly while maintaining the search filter
- Clear the search box
- Verify the table rows return to show all rows matching the status filter
- Repeat the search test typing the `Enter` key instead of clicking the search icon
